code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -567,6 +573,27 @@
 	return TRUE
 
 /datum/reagent/medicine/ephedrine/overdose_process(mob/living/M)
+	if(prob(30) && iscarbon(M))
```

Comment:
```DM
if(iscarbon(M))
    if(your_first_prob)
        your_first_prob_code_here
    if(your_second_prob)
        your_second_prob_code_here
    if(your_third_prob)
        your_third_prob_code_here

if(the_other_prob)
    the_other_prob_code_here
```

Reply:
just fyi this code assumes you wanted all of these to possibly trigger at once.

Reply:
He means the repeated iscarbon checks from line 570 to 585 are uneeded

Reply:
You can avoid going through the probability check only to fail the carbon check x3 if you put the carbon check at the front. I'm fully aware both "work".